core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -351,22 +409,28 @@ private void savePort(DriverChannel channel) {
     }
   }
 
-  private InetSocketAddress getBroadcastRpcAddress(AdminRow row) {
+  private Optional<InetSocketAddress> getBroadcastRpcAddress(AdminRow row) {
+    // in system.peers_v2 (Cassandra >= 4.0)
     InetAddress nativeAddress = row.getInetAddress("native_address");
     if (nativeAddress == null) {
-      // Cassandra < 4
+      // in system.peers or system.local
       InetAddress rpcAddress = row.getInetAddress("rpc_address");
       if (rpcAddress == null) {
-        // This could only happen if system.peers is corrupted, but handle gracefully
-        return null;
+        // This could only happen if system tables are corrupted, but handle gracefully
+        return Optional.empty();
       }
-      return new InetSocketAddress(rpcAddress, port);
+      return Optional.of(new InetSocketAddress(rpcAddress, port));
     } else {
-      Integer rowPort = row.getInteger("native_port");
+      // system.local for Cassandra >= 4.0
+      Integer rowPort = row.getInteger("rpc_port");
```

Comment:
Adding `rpc_port`, seems like an oversight.

Reply:
We're in the branch `nativeAddress != null`, so it's a peers_v2 row. There's no rpc_port column in that table.

Reply:
Right, the method implementation is not correct, I will amend that and simplify.